zerver/lib/push_notifications.py

```diff
@@ -603,6 +603,13 @@ def get_apns_alert_subtitle(message: Message) -> str:
     # For group PMs, or regular messages to a stream, just use a colon to indicate this is the sender.
     return message.sender.full_name + ":"
 
+def get_apns_badge_counts(user_profile: UserProfile) -> int:
+    push_notification_ids = list(UserMessage.objects.filter(
+        user_profile=user_profile,
+        flags=UserMessage.flags.active_mobile_push_notification).order_by(
+            "message_id").values_list("message_id", flat=True))
+    return len(push_notification_ids)
```

Comment:
This can be simplified -- no need to fetch the actual list when we just need the count.

See the discussion of `len()` at the top of this doc:
https://docs.djangoproject.com/en/3.0/ref/models/querysets/